### PR TITLE
[ENH] Improve and add testing to label validation

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Overhaul label validation for classification, regression and anomaly detection to use utility functions

- Unify anomaly detection label checking (@SebastianSchmidl) - main difference is pandas datatypes can accept int instead of just bool and it uses a util function
- disallow single class input for classification (this can still be overridden for cases such as SMOTE, easier to do so now IMO)
- disallow binary inputs for regression (labels with only 0 and 1)

### CODE REVIEW

File: aeon/utils/validation/labels.py [hunk 1]
```diff
@@ -50,21 +60,30 @@ def check_regression_y(y):
...
+    if len(y) == 0:
+        raise ValueError("y must not be empty.")
 
     y_type = type_of_target(y, input_name="y")
-    if y_type != "continuous":
+    if y_type != "continuous" and y_type != "multiclass":
         raise ValueError(
             f"y type is {y_type} which is not valid for regression. "
             f"Should be continuous according to sklearn.utils.multiclass.type_of_target"
         )
```
Comment (TonyBagnall) [inline aeon/utils/validation/labels.py:79]: this is a bit confusing, if its multiclass its ordinal?

Comment (MatthewMiddlehurst) [reply aeon/utils/validation/labels.py:79]: multiclass covers integer values, its a best guess really. impossibly to determie from array input alone. only doing continuous would make [1.0,1.0,1.0,2.0,2.0,2.0,3.0,3.0,3.0] invalid input.

using the method to filter out binary and then manually combing for strings seems sensible to me.

Comment (MatthewMiddlehurst) [reply aeon/utils/validation/labels.py:79]: ordinal is a whole other beast we dont really cover yet I think 🙂

**Review by SebastianSchmidl — APPROVED** [review]
Anomaly detection parts look good 🚢
